app/src/screens/PersonCredentialLoading.tsx

```diff
@@ -0,0 +1,258 @@
+    subscriptions.push(
+      DeviceEventEmitter.addListener(AttestationEventTypes.FailedRequestCredential, handleFailedAttestation)
+    )
+
+    return () => {
+      subscriptions.forEach((subscription) => subscription.remove())
+    }
+  }, [])
+
+  useEffect(() => {
+    // If we are fetching an attestation credential, do no yet have
+    // a remote connection ID to the IAS agent, or the agent is not
+    // initialized, do nothing.
+    if (attestationLoading || !remoteAgentDetails || !agent) {
+      return
+    }
+
+    // We have an attestation credential and can respond to an
+    // attestation proof request.
+    const proofRequest = receivedProofRequests.find((proof) => proof.connectionId === remoteAgentDetails.connectionId)
+    if (!proofRequest) {
+      // No proof from our IAS Agent to respond to, do nothing.
+      return
+    }
+
+    timer.current && clearTimeout(timer.current)
+
+    if (!didCompleteAttestationProofRequest) {
+      acceptAttestationProofRequest(agent, proofRequest)
+        .then((status: boolean) => {
+          // We can unblock the workflow and proceed with
+          // authentication.
+          setDidCompleteAttestationProofRequest(status)
+          logger.info(`Accepted IAS attestation proof request with status: ${status}`)
+        })
+        .catch((error) => {
+          setDidCompleteAttestationProofRequest(false)
+          logger.error(`Unable to accept IAS attestation proof request, error: ${error.message}`)
+        })
+    }
+  }, [attestationLoading, receivedProofRequests, remoteAgentDetails, agent])
+
+  useEffect(() => {
+    if (!remoteAgentDetails || !remoteAgentDetails.legacyConnectionDid || !didCompleteAttestationProofRequest) {
+      return
+    }
+
+    const cb = (status: boolean) => {
+      logger.info(`Service card authentication reported ${status}`)
+      // TODO(jl): Handle the case where the service card authentication fails for
+      // user reasons or otherwise.
+      if (!status) {
+        setDidCompleteAttestationProofRequest(false)
+      }
+    }
+
+    const { iasPortalUrl } = store.developer.environment
+    const { legacyConnectionDid } = remoteAgentDetails
+
+    authenticateWithServiceCard(legacyConnectionDid, iasPortalUrl, cb)
+      .then(() => {
+        logger.error('Completed service card authentication successfully')
+      })
+      .catch((error) => {
+        logger.error('Completed service card authentication with error, error: ', error.message)
+      })
+  }, [remoteAgentDetails, didCompleteAttestationProofRequest])
+
+  useEffect(() => {
+    for (const credential of receivedCredentialOffers) {
+      if (
+        credential.state == CredentialState.OfferReceived &&
+        credential.connectionId === remoteAgentDetails?.connectionId
+      ) {
+        goToCredentialOffer(credential.id)
+      }
+    }
+  }, [receivedCredentialOffers, remoteAgentDetails])
```

Comment:
`store`?

Reply:
how about store.developer.environment.iasPortalUrl so we're not reacting to every change that happens on the store object?